pkg/config/provider.go

```diff
@@ -0,0 +1,59 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/viper"
+)
+
+type IConfigProvider interface {
```

Comment:
This interface has been introduced to provide testable access to the config system for non-interactive usage e.g. when the login command is done we need to set the appropriate config.

At the moment this isn't used by the `octopus config get` or `octopus config set` commands as they have different requirements around prompting for missing keys and special handling of some values.